src/lib.rs

```diff
@@ -225,66 +220,77 @@ macro_rules! _maybe_std_concat_bytes {
 ///   concat_slices!([usize]: /* ... */);
 ///   ```
 ///
-/// - If the type is not a std integer, `f32`, `f64`, or `char` type then you
-///   must also provide an initializer expression.
-///
 ///   ```
 ///   # use constcat::concat_slices;
-///   concat_slices!([(0, 0, 0); (u8, u8, u8)]: /* ... */);
+///   concat_slices!([(u8, u8, u8)]: /* ... */);
 ///   ```
-/// - This also works for custom types as long as the type and initializer
-///   expression is able to be specified in an array initializer expression.
+/// - This also works for custom types as long as the type implement `Copy`.
 ///
 ///   ```
 ///   # use constcat::concat_slices;
 ///   #[derive(Clone, Copy)]
 ///   struct i256(i128, i128);
 ///
-///   impl i256 {
-///       const fn new() -> Self { Self(0, 0) }
-///   }
-///
-///   concat_slices!([i256::new(); i256]: /* ... */);
+///   concat_slices!([i256]: /* ... */);
 ///   ```
 ///
 /// See the [crate documentation][crate] for examples.
 #[macro_export]
 macro_rules! concat_slices {
     ([$init:expr; $T:ty]: $($s:expr),* $(,)?) => {
-        $crate::_concat_slices!([$init; $T]: $($s),*)
+        $crate::_concat_slices!([$T]: $($s),*)
     };
 
+    // Left in for backwards compatiblity.
```

Comment:
Typo, and I think this comment should be on the other rule?

Reply:
I'm more inclined to just remove the old version and release 0.5.0, what do you think?

Reply:
> Typo, and I think this comment should be on the other rule?

Yes, you are correct.

> I'm more inclined to just remove the old version and release 0.5.0, what do you think?

I think that makes more sense. While we could keep the old branch, who knows how long it will stay there without any actual purpose. If that's the direction we want to take I can remove the old one no problem.

Reply:
Cool, let's remove the old version.

Reply:
Done :+1: